src/resolve/metadataResolver.ts

```diff
@@ -71,7 +71,11 @@ export class MetadataResolver {
     const components: SourceComponent[] = [];
     const ignore = new Set();
 
-    if (this.forceIgnore?.denies(dir)) {
+    // don't apply forceignore rules against dirs
+    // `forceignore.denies` will pass a relative path to node-ignore, e.g.
+    // `path/to/force-app` -> `force-app`, note that there's no trailing slash
+    // so node-ignore will treat it as a file.
+    if (!this.tree.isDirectory(dir) && this.forceIgnore?.denies(dir)) {
```

Comment:
@mshanemc it turns out the bug also affects a real FS tree:
![Screenshot 2023-09-01 at 18 51 35](https://github.com/forcedotcom/source-deploy-retrieve/assets/6853656/055f263b-b0ae-4735-a68d-4358d9e1f80f)

`path.relative` doesn't return a trailing slash so `node-ignore` will treat as a file, called here:
https://github.com/forcedotcom/source-deploy-retrieve/blob/f288319e55378d1a7caca5bcd762b77b2c59f30a/src/resolve/forceIgnore.ts#L66C1-L66C1

another option I showed you this morning was to add the trailing slash before passing `dir` to `denies`, but then have to remove it again or it will not be able to file the folder in the tree:
https://github.com/forcedotcom/source-deploy-retrieve/blob/f288319e55378d1a7caca5bcd762b77b2c59f30a/src/resolve/metadataResolver.ts#L78

that also worked, but not sure if it's the right solution:
https://stackoverflow.com/questions/980255/should-a-directory-path-variable-end-with-a-trailing-slash